### PR TITLE
Scala Essentials Kickoff

### PR DESCRIPTION
**Pull request by RHDZMOTA**
#  Description

This PR represents the kickoff of the **Scala Essentials** Nanodegree by Intersys Consulting. 

As discussed in the follow-up session, in this PR the general repository structure and templates are created. The content of the nanodegree should be available in a documentation format and as slides. For both presentations, we use the Markdown markup language. This enables us to focus on writing quality content and use automatization tools such as Pandoc, Scala Tut, Scala Microsite and Jekyll to generate the visually compelling site. 

The content will be published into Github pages when accepted.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
+# Essential Scala Collaborators
+
+Congratulations! You have been selected as a content owner for the  **Scala Essentials** nanodegree.
+
+
+## About the nanodegree
+
+The **Scala Essentials** nanodegree aims to provide a comprehensive tour of the Scala Programming Language for begginers. 
+
+You can see the design document [here](https://docs.google.com/document/d/1HZ1Rsn46x_BVAo8scSyVr9r5IesyTyHhGptjW1n_T6A/edit?usp=sharing).
```
Comment (Rafailong) [inline CONTRIBUTING.md]: Should this document be public?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+7. Idiomatic Scala
+8. Type classes
+9. Functional programming
+10. Popular frameworks and toolkits
+
+## About content owners
+
+As a content owner your job is to design and create the content of at least one module. You can build a small team with other content owners to assist on the content creation. Please consider that each conent owner has full responsibility of their own modules. 
+
+Content owners:
```
Comment (Rafailong) [inline CONTRIBUTING.md]: What about adding a contact method, email maybe?, of every contributor.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+## Requirements
+
+**Note:** Collaborators running on Windows 10 are encoraged to try the [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10) (WSL). 
+
+As a content owner, you'll need the following tools in order to successfuly contribute to the project: 
+
+* The [git](https://git-scm.com/) open source distributed version control system.
+* A [github](http://github.com) account and access to the [Intersys Consulting](https://github.com/intersysconsulting) organization on the platform.
+* At least Java 8 and the Scala Build Tool [SBT](https://www.scala-sbt.org/1.x/docs/Setup.html).
```
Comment (Rafailong) [inline CONTRIBUTING.md]: The most popular way to word this down is: "Java v8 or newer"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+* The [Jekyll](https://jekyllrb.com/) site generator. 
+* [Pandoc](https://pandoc.org/), the file converter tool. 
+
+A setup script on a **debian-based OS** should look similar to:
```
Comment (Rafailong) [inline CONTRIBUTING.md]: Is this script in an actual script file? Maybe having the script file can help a bit more in the setup process.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+            |    |- module3.md
+            |    |- ...
+            |- index.md
+            |- docs.md
+            |- slides.md
+            
+```
+## Development
+
+The development of each module should occur on a separate and individual branch that derives from `develop`
```
Comment (Rafailong) [inline CONTRIBUTING.md]: The branch naming convention remains me to git-flow, if there a plan to follow such a versioning style let's be explicit bout it.

File: README.md [hunk 6]
```diff
@@ -1,18 +1,62 @@
...
+
+Prospectus students must:
+
+* Have their own hardware with the following minimum specs:
+    * At least **4GB of RAM**. See the system requirements for IntelliJ [here](https://bit.ly/1IDJJ7h).
+    * Be able to dedicate **at least 1 hour of individual learning** for each module (homework, readings, etc).
+    
+    
+    
+## What you will learn
```
Comment (Rafailong) [inline README.md]: Always prefer contractions: `...you'll learn`

File: src/main/resources/microsite/slides/reveal.js/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+## Contributing
+
+Please keep the [issue tracker](http://github.com/hakimel/reveal.js/issues) limited to **bug reports**, **feature requests** and **pull requests**.
+
+
+### Personal Support
+If you have personal support or setup questions the best place to ask those are [StackOverflow](http://stackoverflow.com/questions/tagged/reveal.js).
```
Comment (Rafailong) [inline src/main/resources/microsite/slides/reveal.js/CONTRIBUTING.md]: If you need...

File: CONTRIBUTING.md (hunk 1 above)
Comment (RHDZMOTA) [reply CONTRIBUTING.md]: Good catch @Rafailong ! No need for sharing this doc.

File: CONTRIBUTING.md (hunk 2 above)
Comment (RHDZMOTA) [reply CONTRIBUTING.md]: We are thinking about using Gitter as the official communication channel for this Nanodegree (will add the gitter badge soon). Nonetheless, I agree on providing individual contact method for each collaborator.

Comment (RHDZMOTA) [reply CONTRIBUTING.md]: We can use the gitter private messaging url for each collaborator (e.g. https://gitter.im/oscarvarto).

File: CONTRIBUTING.md (hunk 4 above)
Comment (RHDZMOTA) [reply CONTRIBUTING.md]: Okay, I'll create a `setup/debian-based.sh` script with this contents. Maybe a Windows and Mac user can help us with a script for their platforms!

File: CONTRIBUTING.md (hunk 5 above)
Comment (RHDZMOTA) [reply CONTRIBUTING.md]: Makes sense!

File: src/main/resources/microsite/slides/reveal.js/CONTRIBUTING.md (hunk 7 above)
Comment (RHDZMOTA) [reply src/main/resources/microsite/slides/reveal.js/CONTRIBUTING.md]: This is actually a Reveal.js issue!

**Pull request merged**